game/core/src/main/java/com/gg/game/Model/Level.java

```diff
@@ -4,32 +4,94 @@
 import java.util.List;
 
 class Level {
-    private List<Room> rooms;
     private Room currentRoom;
 
-    Level() {
-        rooms=new ArrayList<Room>();
-        rooms.add(new Room(100,100));
+    Level(List<String> connectionList) {
+        loadLevels(connectionList);
     }
 
-    TextureType getCurrentRoomBackground(){
-        return rooms.get(0).getBackground();
+    private void loadLevels(List<String> connectionList) {
+        List<Room> rooms = new ArrayList<>();
+
+        for (String connection : connectionList) {
+            String[] pairs = connection.split(":");
+            String[] pair = pairs[0].split(",");
+            Room r1 = new Room(Integer.parseInt(pair[0]), Integer.parseInt(pair[1]));
+            pair = pairs[1].split(",");
+            Room r2 = new Room(Integer.parseInt(pair[0]), Integer.parseInt(pair[1]));
+
+            if (!rooms.contains(r1)) {
+                rooms.add(r1);
+            }
+
+            if (!rooms.contains(r2)) {
+                rooms.add(r2);
+            }
+
+            int dx = r2.getX() - r1.getX();
+            int dy = r2.getY() - r1.getY();
+
+            if (dx == -1) {
+                rooms.get(rooms.indexOf(r1)).setWest(rooms.get(rooms.indexOf(r2)));
+            } else if (dx == 1) {
+                rooms.get(rooms.indexOf(r1)).setEast(rooms.get(rooms.indexOf(r2)));
+            } else if (dy == -1) {
+                rooms.get(rooms.indexOf(r1)).setSouth(rooms.get(rooms.indexOf(r2)));
+            } else if (dy == 1) {
+                rooms.get(rooms.indexOf(r1)).setNorth(rooms.get(rooms.indexOf(r2)));
+            }
+        }
+
+        currentRoom = rooms.get(0);
     }
 
+    protected void goNorth() {
```

Comment:
All of these four ```goNorth()```, ```goSouth()```, ```goEast()``` and ```goWest()``` can be replaced by a generic method of some sort as they share their code near completely.

```suggestion
    private enum CardinalDirection { NORTH, SOUTH, EAST, WEST }
    
    protected void goToRoom(CardinalDirection direction) {
        ....
    }
```
Would that be preferable?

Mind you that this might not be the right approach if using Graph as suggested in Room.java.